.github/workflows/gh-build-docs.yml

```diff
@@ -0,0 +1,162 @@
+          miniforge-version: latest
+          conda-remove-defaults: "true"
+          python-version: 3.12
+
+      - name: Check conda env
+        run: |
+          conda info
+          conda list
+          conda config --show-sources
+          conda config --show
+
+      # WAR: Building the doc currently requires CTK installed (NVIDIA/cuda-python#326,327)
+      - name: Set up mini CTK
+        uses: ./.github/actions/fetch_ctk
+        continue-on-error: false
+        with:
+          host-platform: linux-64
+          cuda-version: ${{ inputs.build_ctk_ver }}
+
+      - name: Set environment variables
+        run: |
+          PYTHON_VERSION_FORMATTED="312"  # see above
+          REPO_DIR=$(pwd)
+
+          # make outputs from the previous job as env vars
+          echo "CUDA_CORE_ARTIFACT_NAME=cuda-core-python${PYTHON_VERSION_FORMATTED}-linux-64-${{ github.sha }}" >> $GITHUB_ENV
+          echo "CUDA_CORE_ARTIFACTS_DIR=$(realpath "$REPO_DIR/cuda_core/dist")" >> $GITHUB_ENV
+          echo "CUDA_BINDINGS_ARTIFACT_NAME=cuda-bindings-python${PYTHON_VERSION_FORMATTED}-cuda${{ inputs.build_ctk_ver }}-linux-64-${{ github.sha }}" >> $GITHUB_ENV
+          echo "CUDA_BINDINGS_ARTIFACTS_DIR=$(realpath "$REPO_DIR/cuda_bindings/dist")" >> $GITHUB_ENV
+
+      - name: Download cuda.bindings build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ env.CUDA_BINDINGS_ARTIFACT_NAME }}
+          path: ${{ env.CUDA_BINDINGS_ARTIFACTS_DIR }}
+
+      - name: Display structure of downloaded cuda.bindings artifacts
+        run: |
+          pwd
+          ls -lahR $CUDA_BINDINGS_ARTIFACTS_DIR
+
+      - name: Download cuda.core build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ env.CUDA_CORE_ARTIFACT_NAME }}
+          path: ${{ env.CUDA_CORE_ARTIFACTS_DIR }}
+
+      - name: Display structure of downloaded cuda.core build artifacts
+        run: |
+          pwd
+          ls -lahR $CUDA_CORE_ARTIFACTS_DIR
+
+      - name: Install all packages
+        run: |
+          pushd "${CUDA_BINDINGS_ARTIFACTS_DIR}"
+          pip install *.whl
+          popd
+
+          pushd "${CUDA_CORE_ARTIFACTS_DIR}"
+          pip install *.whl
+          popd
+
+      - name: Build all (latest) docs
+        id: build
+        run: |
+          pushd cuda_python/docs/
+          ./build_all_docs.sh latest-only
+          ls -l build
+          popd
+
+          mkdir -p artifacts/docs
+          mv cuda_python/docs/build/html/* artifacts/docs/
+
+      # Note: currently this is only for manual inspection. This step will become
+      # required once we switch to use GHA for doc deployment (see the bottom).
+      - name: Upload doc artifacts
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: artifacts/
+          retention-days: 3
+
+      # The steps below are not executed unless when building on main.
```

Comment:
Note: The steps below can only be verified once this PR is merged... For reference, see the same deployment steps used in the array-api repo: https://github.com/data-apis/array-api/blob/main/.github/workflows/pages.yml

Reply:
Update: In commit 45218b861cedf2cd8f327d43129e2b6ab48ef704 I switched to use a popular action `github-pages-deploy-action` for this purpose. It has been approved for use in the NVIDIA org, so at least one other team has started using it.